config/ingest.go

```diff
@@ -6,37 +6,56 @@ import (
 
 // Ingest tracks the configuration related to the ingestion protocol.
 type Ingest struct {
+	// The recursion depth limit when syncing advertisements. The value -1
+	// means no limit and zero means use the default value. This prupose is to
+	// prevent overload from endless advertisement chains.
+	AdvertisementDepthLimit int
+	// The recursion depth limit when syncing advertisement entries. The value
+	// -1 means no limit and zero means use the default value. This prupose is
+	// to prevent overload from endless entries chains.
+	EntriesDepthLimit int
+	// How many ingest worker goroutines to spawn. This controls how many
+	// concurrent ingest from different providers we can handle.
+	IngestWorkerCount int
 	// PubSubTopic used to advertise ingestion announcements.
 	PubSubTopic string
 	// StoreBatchSize is the number of entries in each write to the value
-	// store.  Specifying a value less than 2 disables batching.  This should
+	// store.  Specifying a value less than 2 disables batching. This should
 	// be smaller than the maximum number of multihashes in an entry block to
 	// write concurrently to the value store.
 	StoreBatchSize int
 	// SyncTimeout is the maximum amount of time allowed for a sync to complete
 	// before it is canceled. This can be a sync of a chain of advertisements
-	// or a chain of advertisement entries.  The value is an integer string
+	// or a chain of advertisement entries. The value is an integer string
 	// ending in "s", "m", "h" for seconds. minutes, hours.
 	SyncTimeout Duration
-	// How many ingest worker goroutines to spawn. This controls how many
-	// concurrent ingest from different providers we can handle.
-	IngestWorkerCount int
 }
 
 // NewIngest returns Ingest with values set to their defaults.
 func NewIngest() Ingest {
 	return Ingest{
-		PubSubTopic:       "/indexer/ingest/mainnet",
-		StoreBatchSize:    4096,
-		SyncTimeout:       Duration(2 * time.Hour),
-		IngestWorkerCount: 10,
+		AdvertisementDepthLimit: 65536,
```

Comment:
some providers will pretty quickly exceed this, i think

Reply:
If a publisher publishes an advertisement every second, then that is 86400 a day, 604800 a week, 31.5M a year. So maybe something > 32M?

I can use a large value above or I could make it unlimited by default (-1), but still have it configurable if someone needs it. WDYT?

Reply:
32M seems pretty safe